src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -402,6 +402,37 @@ private static bool TryGetClassDeclarationList(INamedTypeSymbol typeSymbol, [Not
                 return typeGenerationSpec;
             }
 
+            internal static bool IsSyntaxTargetForGeneration(SyntaxNode node) => node is ClassDeclarationSyntax { AttributeLists: { Count: > 0 }, BaseList: { Types : {Count : 0 } } };
+
+            internal static ClassDeclarationSyntax? IsSemanticTargetForGeneration(GeneratorSyntaxContext context)
```

Comment:
I would rename this to `GetSemanticTargetForGeneration`. Having a method named "IsXXX" and not return a `bool` seems odd.